python/elastic_search_serialization.py

```diff
@@ -0,0 +1,505 @@
+    )
+    print(f"Client info: {client.info()}")
+    return client
+
+
+def create_node_index(
+    client: Elasticsearch,
+    index_name: str = "node_index",
+    number_of_shards: int = 1,
+    number_of_replicas: int = 1,
+    analyzer: str = "lk_analyzer",
+) -> None:
+    """
+    Creates node index if it doesn't exist before.
+    Args:
+        client (Elasticsearch): A reference to the Elasticsearch client.
+        index_name (str): Name you want to give to the node index. Default one is 'node_index'.
+        number_of_shards (int): A number of shards you want to use in your index.
+        number_of_replicas (int): A number of replicas you want to use in your index.
+    """
+    global node_index_body
+    node_index_body[SETTINGS][INDEX][NUMBER_OF_SHARDS] = number_of_shards
+    node_index_body[SETTINGS][INDEX][NUMBER_OF_REPLICAS] = number_of_replicas
+    node_index_body[MAPPINGS][DYNAMIC_TEMPLATES][0][LK_CATEGORIES_HAS_RAW][MAPPING][ANALYZER] = analyzer
+    node_index_body[MAPPINGS][DYNAMIC_TEMPLATES][1][STRING][MAPPING][ANALYZER] = analyzer
+    client.indices.create(index=index_name, body=node_index_body, ignore=400)
+
+
+def create_edge_index(
+    client: Elasticsearch,
+    index_name: str = "edge_index",
+    number_of_shards: int = 1,
+    number_of_replicas: int = 1,
+    analyzer: str = "lk_analyzer",
+) -> None:
+    """
+    Creates edge index if it doesn't exist before.
+    Args:
+        client (Elasticsearch): A reference to the Elasticsearch client.
+        index_name (str): Name you want to give to the edge index. Default one is 'edge_index'.
+        number_of_shards (int): A number of shards you want to use in your index.
+        number_of_replicas (int): A number of replicas you want to use in your index.
+    """
+    global edge_index_body
+    edge_index_body[SETTINGS][INDEX][NUMBER_OF_SHARDS] = number_of_shards
+    edge_index_body[SETTINGS][INDEX][NUMBER_OF_REPLICAS] = number_of_replicas
+    edge_index_body[MAPPINGS][DYNAMIC_TEMPLATES][0][LK_TYPE_HAS_RAW][MAPPING][ANALYZER] = analyzer
+    edge_index_body[MAPPINGS][DYNAMIC_TEMPLATES][1][STRING][MAPPING][ANALYZER] = analyzer
+    client.indices.create(index=index_name, body=edge_index_body, ignore=400)
+
+
+def generate_documents(
+    context_vertices: List[mgp.Vertex], context_edges: List[mgp.Edge]
+) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]]]:
+    """
+    Generates nodes and edges documents for indexing and returns them as lists.
+    Args:
+        context_vertices (List[mgp.Vertex]]): Vertices in Memgraph that were created/updated.
+        context_edges (List[mgp.Edge]]): Edges in Memgraph that were created/updated.
+    Returns:
+        Tuple[List[Dict[str, Any]], List[Dict[str, Any]]]
+    """
+    nodes, edges = [], []
+    for vertex in context_vertices:
+        nodes.append(serialize_vertex(vertex))
+    for edge in context_edges:
+        edges.append(serialize_edge(edge))
+    return nodes, edges
```

Comment:
Can we parallelize this? Does it make sense?

Reply:
Hmm not sure I think yes. Multiprocessing in Python. But this requires rewriting because I don't think setting up more than a multiprocessing pool makes sense.